File: Fe/src/features/auth/services/auth.service.ts
import { apiRequest, tokenStore } from '../../../services/apiClient'
import { endpoints } from '../../../services/endpoints'
import type { AuthTokens, LoginCredentials, ChangePasswordData } from '../models/auth.model'

export async function login(credentials: LoginCredentials) {
  const tokens = await apiRequest<AuthTokens>(endpoints.login, { method: 'POST', body: JSON.stringify(credentials) })
  tokenStore.save(tokens.accessToken, tokens.refreshToken)
}

export async function logout() {
  const refreshToken = tokenStore.refresh()
  try { if (refreshToken) await apiRequest<void>(endpoints.logout, { method: 'POST', body: JSON.stringify({ refreshToken }) }, false) }
  finally { tokenStore.clear() }
}

export async function changePassword(passwordData: ChangePasswordData) {
  await apiRequest<void>(endpoints.changePassword, { method: 'POST', body: JSON.stringify(passwordData) })
}

export async function logoutAll() {
  await apiRequest<void>(endpoints.logoutAll, { method: 'POST' })
  tokenStore.clear()
}

File: TodoListProject/src/main/java/com/project/taskmanagement/service/AuthService.java
package com.project.taskmanagement.service;

import com.project.taskmanagement.dto.request.auth.ChangePasswordRequest;
import com.project.taskmanagement.dto.request.auth.LoginRequest;
import com.project.taskmanagement.dto.response.auth.AuthResponse;

public interface AuthService {

    AuthResponse login(LoginRequest request);

    AuthResponse refresh(String refreshToken);

    void logout(String accessToken, String refreshToken);

    void changePassword(String username, ChangePasswordRequest request);

    void logoutAll(String accessToken);
}

